lib/nfg/capistrano/tasks/deploy.rake

```diff
@@ -0,0 +1,39 @@
+require 'aws-sdk-s3'
+
+namespace :deploy do
+  Rake::Task['deploy:compile_assets'].clear
+
+  desc 'Download or compile assets'
+  task :compile_assets => [:set_rails_env] do
+    assets_cache_prefix = Nfg::Capistrano::Config.circleci[:parameters][:assets_cache_prefix][:default]
+    revision = %x(git --no-pager log -1 --pretty=format:%H)
+    branch = fetch(:branch).gsub('/', '_')
+    assets_filename="#{assets_cache_prefix}-assets-#{revision}-#{branch}.tar.gz"
+    s3_bucket = Nfg::Capistrano::Config.circleci[:parameters][:s3_bucket][:default]
+    aws_credentials = Aws::Credentials.new(
+      Nfg::Capistrano::Config.api_keys[:aws_access_key_id],
+      Nfg::Capistrano::Config.api_keys[:aws_secret_access_key]
+    )
+    s3_assets_file = Aws::S3::Object.new(
+      region: ENV.fetch('AWS_REGION', 'us-east-1'),
+      credentials: aws_credentials,
+      bucket_name: s3_bucket.gsub('s3://', ''),
+      key: "assets/#{assets_filename}"
+    )
+
+    if s3_assets_file.exists?
+      s3.head_object(bucket: s3_bucket.gsub('s3://', ''), key: "assets/#{assets_filename}")
```

Comment:
I think this is redundant now unless I'm mistaken.

Reply:
Thank you. Fixed.